src/main/java/entities/Budget.java

```diff
@@ -10,9 +14,10 @@
  * A budget is associated with the length of time for which the budget is active and a limit on the amount on money that
  * can be spent on items in the budget.
  */
-public class Budget {
+public class Budget implements VetoableChangeListener, PropertyChangeListener {
```

Comment:
VetoableChangeListener is like PropertyChangeListener, except a listener can "veto" a change by throwing an exception